docs/agent_api.rst

```diff
@@ -3,4 +3,334 @@
+
+There are as many methods available as existing DDS entities, replacing the parameters *<entity>* and *<associated_entity>* as shown above in the previous method description (see :ref:`existing_entities_and_associated_entities`).
+
+This operation returns `true` if the entity is successfully created and linked to its associated entity (which must previously exist in the given `ProxyClient`), or false otherwise.
+
+:client_key: The identifier of the `ProxyClient` to which the resulting entity will be attached to.
+:<entity>_id: The ID of the DDS entity to be created.
+:<associated_entity>_id: The identifier of the DDS entity to which this one will be assocciated.
+:ref: The reference tag which will retrieve the DDS entity description from the references file.
+:flag: It determines the creation mode of the new entity (see :ref:`creation_mode_table`).
+:op_result: The result status of this operation.
+
+..
+    TODO: Add reference to creation mode section (xml/ref).
+
+------
+
+.. code-block:: cpp
+
+    bool delete_<entity>(uint32_t client_key, uint16_t <entity>_id, OpResult& op_result);
+
+Deletes a certain entity from a `ProxyClient`. Its associated entities will also be deleted, if applicable.
+
+There exist as many method signatures of this type in the agent's API as available entities. See the :ref:`existing_entities_and_associated_entities` table for further information.
+
+Returns `true` if the entity is correctly removed, or `false` otherwise.
+
+:client_key: The identifier of the `ProxyClient` from which the entity must be deleted.
+:<entity>_id: The ID of the DDS entity to be deleted.
+:op_result: The result status of the operation.
+
+------
+
+.. code-block:: cpp
+
+    bool load_config_file(const std::string& file_path);
+
+Loads a configuration file which provides with the tagged XML definitions of the desired XRCE entities that can be created using the reference creation mode.
+
+The used syntax must match the one defined for [FastDDS XML profile syntax](https://fast-dds.docs.eprosima.com/en/latest/fastdds/xml_configuration/xml_configuration.html),
+where the *profile name* attributes represent the reference's names.
+
+Returns `true` if the file was correctly loaded or `false` otherwise.
+
+..
+    TODO: Add reference to creation mode section (xml/ref).
+
+:file_path: Relative path to the references file.
+
+------
+
+.. code-block:: cpp
+
+    void reset();
+
+Deletes all the `ProxyClient` instances and their associated DDS entities.
+
+------
+
+.. code-block:: cpp
+
+    void set_verbose_level(uint8_t verbose_level);
+
+Sets the verbose level of the logger, from `0` (logger is off) to `6` (critical, error, warning, info, debug and trace messages displayed).
+
+:verbose_level: The level to be set.
+
+------
+
+.. code-block:: cpp
+
+    template <typename ... Args>
```

Comment:
Isn't this repeated from line 69?

Reply:
Yes, kind of. The thing is that both `AgentInstance` and `Agent` can invoke this method, so... maybe I could reference one to another